File: app/src/test/java/com/nytimes/android/external/register/APIOverridesTest.java
package com.nytimes.android.external.register;

import android.content.Context;
import android.content.SharedPreferences;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.MockitoAnnotations;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.RuntimeEnvironment;

import static com.nytimes.android.external.register.APIOverrides.PREF_NAME;
import static org.assertj.core.api.Assertions.assertThat;

@RunWith(RobolectricTestRunner.class)
public class APIOverridesTest {

    private static final int IS_BILLING_RESPONSE = -100;
    private static final int GET_BUY_INTENT_RESPONSE = -101;
    private static final int BUY_RESPONSE = -102;
    private static final int GET_PURCHASES_RESPONSE = -103;
    private static final int GET_SKU_DETAILS_RESPONSE = -104;
    private static final int CONSUME_PURCHASE_RESPONSE = -105;
    private static final String GET_USERS_RESPONSE = "myfun@user.com";
    private APIOverrides testObject;

    @Before
    public void setUp() {
        MockitoAnnotations.initMocks(this);

        SharedPreferences sharedPreferences =
                RuntimeEnvironment.application.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);

        testObject = new APIOverrides(sharedPreferences);
    }

    @Test
    public void testIsBillingSupported() {
        testObject.setIsBillingSupportedResponse(IS_BILLING_RESPONSE);
        assertThat(testObject.getIsBillingSupportedResponse())
                .isEqualTo(IS_BILLING_RESPONSE);
    }

    @Test
    public void testBuyIntentResponse() {
        testObject.setGetBuyIntentResponse(GET_BUY_INTENT_RESPONSE);
        assertThat(testObject.getGetBuyIntentResponse())
                .isEqualTo(GET_BUY_INTENT_RESPONSE);
    }

    @Test
    public void testBuyResponse() {
        testObject.setBuyResponse(BUY_RESPONSE);
        assertThat(testObject.getBuyResponse())
                .isEqualTo(BUY_RESPONSE);
    }

    @Test
    public void testPurchasesResponse() {
        testObject.setGetPurchasesResponse(GET_PURCHASES_RESPONSE);
        assertThat(testObject.getGetPurchasesResponse())
                .isEqualTo(GET_PURCHASES_RESPONSE);
    }

    @Test
    public void testGetSkuDetailsResponse() {
        testObject.setGetSkuDetailsResponse(GET_SKU_DETAILS_RESPONSE);
        assertThat(testObject.getGetSkuDetailsResponse())
                .isEqualTo(GET_SKU_DETAILS_RESPONSE);
    }

    @Test
    public void testConsumePurchaseResponse() {
        testObject.setConsumePurchaseResponse(CONSUME_PURCHASE_RESPONSE);
        assertThat(testObject.getConsumePurchaseResponse())
                .isEqualTo(CONSUME_PURCHASE_RESPONSE);
    }

    @Test
    public void testGetUsersResponse() {
        testObject.setUsersReponse(GET_USERS_RESPONSE);
        assertThat(testObject.getUsersResponse())
                .isEqualTo(GET_USERS_RESPONSE);
    }
}

File: app/src/test/java/com/nytimes/android/external/register/APIOverridesTest.kt
package com.nytimes.android.external.register

import android.content.Context
import com.nytimes.android.external.register.APIOverrides.PREF_NAME
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.RuntimeEnvironment

@RunWith(RobolectricTestRunner::class)
class APIOverridesTest {

    private lateinit var testObject: APIOverrides

    @Before
    fun setUp() {
        val sharedPreferences = RuntimeEnvironment.application
                .getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)

        testObject = APIOverrides(sharedPreferences)
    }

    @Test
    fun testIsBillingSupported() {
        testObject.isBillingSupportedResponse = IS_BILLING_RESPONSE
        assertThat(testObject.isBillingSupportedResponse)
                .isEqualTo(IS_BILLING_RESPONSE)
    }

    @Test
    fun testBuyIntentResponse() {
        testObject.getBuyIntentResponse = GET_BUY_INTENT_RESPONSE
        assertThat(testObject.getBuyIntentResponse)
                .isEqualTo(GET_BUY_INTENT_RESPONSE)
    }

    @Test
    fun testBuyResponse() {
        testObject.buyResponse = BUY_RESPONSE
        assertThat(testObject.buyResponse)
                .isEqualTo(BUY_RESPONSE)
    }

    @Test
    fun testPurchasesResponse() {
        testObject.getPurchasesResponse = GET_PURCHASES_RESPONSE
        assertThat(testObject.getPurchasesResponse)
                .isEqualTo(GET_PURCHASES_RESPONSE)
    }

    @Test
    fun testGetSkuDetailsResponse() {
        testObject.getSkuDetailsResponse = GET_SKU_DETAILS_RESPONSE
        assertThat(testObject.getSkuDetailsResponse)
                .isEqualTo(GET_SKU_DETAILS_RESPONSE)
    }

    @Test
    fun testConsumePurchaseResponse() {
        testObject.consumePurchaseResponse = CONSUME_PURCHASE_RESPONSE
        assertThat(testObject.consumePurchaseResponse)
                .isEqualTo(CONSUME_PURCHASE_RESPONSE)
    }

    @Test
    fun testGetUsersResponse() {
        testObject.setUsersReponse(GET_USERS_RESPONSE)
        assertThat(testObject.usersResponse)
                .isEqualTo(GET_USERS_RESPONSE)
    }

    companion object {
        private const val IS_BILLING_RESPONSE = -100
        private const val GET_BUY_INTENT_RESPONSE = -101
        private const val BUY_RESPONSE = -102
        private const val GET_PURCHASES_RESPONSE = -103
        private const val GET_SKU_DETAILS_RESPONSE = -104
        private const val CONSUME_PURCHASE_RESPONSE = -105
        private const val GET_USERS_RESPONSE = "myfun@user.com"
    }
}
